### ISSUE TITLE
Bus error

### ISSUE THREAD

**Issue by Rudloff** [wkhtmltopdf/wkhtmltopdf#3306]
Hello,

I am trying to run wkhtmltopdf 0.12 on an x86 Linux shared hosting and it seems to crash immediately with a bus error.
If I use strace, I get this output:
```
./wkhtmltopdf", ["./wkhtmltopdf", "--version"], [/* 4 vars */]) = 0
brk(0)                                  = 0xa7e1f6c
--- SIGBUS (Bus error) @ 0 (0) ---
+++ killed by SIGBUS +++
```

**Comment by PhilterPaper** [comment]
Usually when you have an invalid hardware address (pointer), you'll get a Segmentation Fault if it points to someone else's storage. However, a physically impossible address can deliver a Bus error instead. The "@ 0 (0)" part of the message suggests this may be address 0, which I've usually seen with uninitialized pointers, but as a SegFault. Maybe your system delivers a Bus error on a 0 pointer instead. Anyway, I hope that might give you a lead as to where to start looking. Have you tried a fresh copy of the source and a rebuild, to rule out some corruption somewhere? Is there a prebuilt version available that works correctly?

**Comment by Rudloff** [comment]
I didn't build it from source. I'm using this binary: http://download.gna.org/wkhtmltopdf/0.12/0.12.4/wkhtmltox-0.12.4_linux-generic-i386.tar.xz
The same binary works perfectly on other computers.
Do you think building from source could fix the problem (I can't build directly on the server itself however).

**Comment by PhilterPaper** [comment]
It's not uncommon for binary executables built on one machine to _not_ work on another machine. For servers, especially, hosts tend to deliberately discourage installation of arbitrary programs due to load on the system. You probably have a mismatch of libraries or something. If you can't build _on_ the server, there's a good chance it won't work (unless the host is very cooperative about exactly what library support is available). Maybe if you link with all static libraries, it can work.

**Comment by Tomsgu** [comment]
Have you resolved the issue? Does `0.12.5` build works for you?

**Comment by Rudloff** [comment]
Sorry, I don't have access to this host anymore so I can't try to reproduce the issue with 0.12.5.
